app/client/api/apiClient.js

```diff
@@ -0,0 +1,49 @@
+import {generateFormattedURL, getTFAToken, getToken, handleResponse} from "../utils";
+
+/**
+ * Sempo apiClient
+ *
+ * @param url, STRING to be queried
+ * @param method, ENUM of available HTTP codes
+ * @param isAuthed, BOOLEAN,
+ * @param isTFA, BOOLEAN,
+ * @param isForm, BOOLEAN,
+ * @param query, OBJECT of key/value pairs to be parsed into a query string
+ * @param body, OBJECT, body data to send to server
+ * @param path, INTEGER, used for calling specific object ID i.e. /1/
+ * @param errorHandling, BOOLEAN, only use FALSE for special use case for TFA/auth
+ * @returns {Promise<Response>}
+ */
+export const apiClient = ({url, method=method.toUpperCase(), isAuthed=true, isTFA=false, isForm=false, query=null, body=null, path=null, errorHandling=true}) => {
+  if (['PUT', 'POST', 'GET'].indexOf(method) === -1) {
+    throw Error('Method provided is not supported')
+  }
+
+  let formData;
+  let headers = {};
+  let request = {
+    headers: headers,
+    method: method,
+  };
+
+  //todo: check which headers are needed for given method
+  if (isForm) {
+    formData = new FormData();
+    Object.keys(body).map(key => {
+      formData.append(key.toString(), body[key]);
+    });
+    method !== 'GET' ? request['body'] = formData : null;
```

Comment:
if we're anticipating a lot more differences in request parameters between GETs, POSTs, etc and adding other request types such as PATCH we might want to have the client return an object of methods and handling corresponding elements inside each function so that, for example, we don't need to process formdata if the request is going to be a GET. 

`return {
     get: () => ....,
     post: () => ....
}`

similar to your thoughts above. But, this is an incredibly minor tweak that can be handled when we start adding more request types